src/main/java/edu/kis/powp/jobs2d/drivers/MacroDriverComposite.java

```diff
@@ -8,7 +8,7 @@
 import java.util.ArrayList;
 import java.util.List;
 
-public class MacroDriverDecorator implements Job2dDriver {
+public class MacroDriverComposite implements Job2dDriver {
```

Comment:
Sorry, but it is still the decorator. You should add DriverComposite class, and MacroDriver should be stripped from a decorator pattern.